legacy/firmware/crypto.c

```diff
@@ -508,6 +508,22 @@ int cryptoIdentityFingerprint(const IdentityType *identity, uint8_t *hash) {
   return 1;
 }
 
+static bool check_cointype(const CoinInfo *coin, uint32_t slip44, bool full) {
+#if BITCOIN_ONLY
+  (void)full;
+#else
+  if (!full) {
```

Comment:
So `full` is `true` for `fsm_msgGetAddress` and `false` for `compile_input_script_sig` in `signing.c`. So we are doing this check for signing only and not for get address. So this disables getting a BCH address on a BTC path. Correct? Is that intended? If we allow signing shouldn't we allow getting an address as well? Also this is different than TT then isn't it?

Reply:
> So this disables getting a BCH address on a BTC path. 

GetAddress is still allowed, but the dialog will show extra warning.

Reply:
I am a bit confused, do we have these checks for legacy for GetAddress? Because `trezorctl get-address -c Bcash -t address -n "m/44'/0'/0'/0/0"` seem to work just fine.

Reply:
Try `trezorctl get-address -c Bcash -t address -n "m/44'/0'/0'/0/0" -d` - the warning is shown on the display - if display is not used, nothing is changed

Reply:
I have created https://github.com/trezor/trezor-firmware/issues/1206 for this because I think the warning should still be there. We can merge this one though.

Reply:
OK, thanks!